### ISSUE TITLE
Tabular files with a single row don't show up properly in data_sets 1  file browser

### ISSUE THREAD

**Issue by scottx611x** [parklab/refinery-platform#1463]
* Specific code commit: fad81a20e6ce1334cde6c3350a1bf89fa4b82a81
* Version of the web browser and OS: `Chrome Version 53.0.2785.143 (64-bit)`
* Environment where the error occurred (Vagrant VM and site conf mode or AWS instance): https://test.stemcellcommons.org

### Steps to reproduce
1. Upload this comma delimited file (or any other tabular file referencing a single data file) into Refinery: [sample-comma.txt](https://github.com/parklab/refinery-platform/files/523334/sample-comma.txt)

2. View the File Browser at `https://test.stemcellcommons.org/data_sets/<uuid>`:
  * See no files displayed (no console errors): <img width="1440" alt="screen shot 2016-10-11 at 9 43 28 pm" src="https://cloud.githubusercontent.com/assets/5629547/19294613/35404bd4-8ffc-11e6-9209-3e6638b89bea.png">

3. View the File Browser at `https://test.stemcellcommons.org/data_sets2/<uuid>`:
  * See the proper behavior: 
<img width="1440" alt="screen shot 2016-10-11 at 9 43 45 pm" src="https://cloud.githubusercontent.com/assets/5629547/19294706/0af09f04-8ffd-11e6-8c26-797b9f0a55e3.png">

### Observed behavior
* I haven't exhausted all edge cases but it seems that tabular files that reference a single data file, regardless of delimiter type,  aren't displayed properly in the `data_sets` file broswer, but are displayed properly in `data_sets2` file browser

### Expected behavior
* For the `data_sets` file browser to be able to handle tabular files referencing a single data file.

**Comment by scottx611x** [comment]
I'm currently not able to reproduce this issue locally in `dev`.

**See:**
![screen shot 2016-10-14 at 1 15 59 pm](https://cloud.githubusercontent.com/assets/5629547/19396303/7563a7ea-9210-11e6-8426-ab50a4a1406a.png)

**Comment by scottx611x** [comment]
But I _am_ able to see the issue in `prod` mode

**See:**
![screen shot 2016-10-14 at 1 17 25 pm](https://cloud.githubusercontent.com/assets/5629547/19396354/ba54da18-9210-11e6-9438-1576b97199ec.png)

**Comment by scottx611x** [comment]
Relevant Apache log:

```
[Fri Oct 14 13:20:49.588615 2016] [:error] [pid 4970:tid 139776429516544] 2016-10-14 13:20:49 INFO     requests.packages.urllib3.connectionpool:203 _new_conn() - Starting new HTTP connection (1): localhost
[Fri Oct 14 13:20:49.592677 2016] [:error] [pid 4970:tid 139776429516544] 2016-10-14 13:20:49 DEBUG    requests.packages.urllib3.connectionpool:383 _make_request() - "GET /solr/data_set_manager/select?fq=%28study_uuid%3A8576a656-dd1e-4192-8e78-ebd14b0125c6+AND+assay_uuid%3Adb27bb7d-2859-49b8-a84f-a7fa59c7bfcb%29&fq=type%3A%28%22Raw+Data+File%22+OR+%22Derived+Data+File%22+OR+%22Array+Data+File%22+OR+%22Derived+Array+Data+File%22+OR+%22Array+Data+Matrix+File%22+OR+%22Derived+Array+Data+Matrix+File%22%29&fq=is_annotation%3Afalse&rows=1&json.wrf=jQuery21103308266235968844_1476465648551&q=django_ct%3Adata_set_manager.node&start=0&wt=json&_=1476465648552 HTTP/1.1" 200 None
[Fri Oct 14 13:20:49.604417 2016] [:error] [pid 4970:tid 139776513443584] 2016-10-14 13:20:49 INFO     requests.packages.urllib3.connectionpool:203 _new_conn() - Starting new HTTP connection (1): localhost
[Fri Oct 14 13:20:49.616089 2016] [:error] [pid 4970:tid 139776513443584] 2016-10-14 13:20:49 DEBUG    requests.packages.urllib3.connectionpool:383 _make_request() - "GET /solr/data_set_manager/select?fq=%28study_uuid%3A8576a656-dd1e-4192-8e78-ebd14b0125c6+AND+assay_uuid%3Adb27bb7d-2859-49b8-a84f-a7fa59c7bfcb%29&fq=type%3A%28%22Raw+Data+File%22+OR+%22Derived+Data+File%22+OR+%22Array+Data+File%22+OR+%22Derived+Array+Data+File%22+OR+%22Array+Data+Matrix+File%22+OR+%22Derived+Array+Data+Matrix+File%22%29&fq=is_annotation%3Atrue&rows=1&json.wrf=jQuery21103308266235968844_1476465648553&q=django_ct%3Adata_set_manager.node&start=0&wt=json&_=1476465648554 HTTP/1.1" 200 None
[Fri Oct 14 13:20:49.634091 2016] [:error] [pid 4970:tid 139776521836288] 2016-10-14 13:20:49 INFO     requests.packages.urllib3.connectionpool:203 _new_conn() - Starting new HTTP connection (1): localhost
[Fri Oct 14 13:20:49.730978 2016] [:error] [pid 4970:tid 139776513443584] 2016-10-14 13:20:49 INFO     requests.packages.urllib3.connectionpool:203 _new_conn() - Starting new HTTP connection (1): localhost
[Fri Oct 14 13:20:49.816207 2016] [:error] [pid 4970:tid 139776521836288] 2016-10-14 13:20:49 DEBUG    requests.packages.urllib3.connectionpool:383 _make_request() - "GET /solr/data_set_manager/select?sort=REFINERY_SUBANALYSIS_2257_953_s+asc&fq=%28study_uuid%3A8576a656-dd1e-4192-8e78-ebd14b0125c6+AND+assay_uuid%3Adb27bb7d-2859-49b8-a84f-a7fa59c7bfcb%29&fq=type%3A%28%22Raw+Data+File%22+OR+%22Derived+Data+File%22+OR+%22Array+Data+File%22+OR+%22Derived+Array+Data+File%22+OR+%22Array+Data+Matrix+File%22+OR+%22Derived+Array+Data+Matrix+File%22%29&fq=is_annotation%3Afalse&rows=20&facet.field=REFINERY_SUBANALYSIS_2257_953_s&facet.field=REFINERY_ANALYSIS_UUID_2257_953_s&facet.field=REFINERY_WORKFLOW_OUTPUT_2257_953_s&facet.limit=-1&facet.sort=count&facet=true&json.wrf=jQuery21103308266235968844_1476465648551&q=django_ct%3Adata_set_manager.node&start=0&wt=json&facet.pivot=%2C&fl=REFINERY_SUBANALYSIS_2257_953_s%2Cspecies_Characteristics_2257_953_s%2Cuuid%2Cname%2Cassay_uuid%2CREFINERY_TYPE_2257_953_s%2Cfile_uuid%2Curl_Characteristics_2257_953_s%2Cid_Characteristics_2257_953_s%2Cis_annotation%2Cstudy_uuid%2CREFINERY_FILETYPE_2257_953_s%2CREFINERY_ANALYSIS_UUID_2257_953_s%2Ctype%2CREFINERY_WORKFLOW_OUTPUT_2257_953_s%2CREFINERY_NAME_2257_953_s&_=1476465648555 HTTP/1.1" 400 None
[Fri Oct 14 13:20:49.817312 2016] [:error] [pid 4970:tid 139776521836288] 2016-10-14 13:20:49 ERROR    core.views:727 solr_select() - 400 Client Error: Bad Request
```

**Comment by scottx611x** [comment]
Solr sends back an additional 400 error in the data_sets 1 filebrowser when there is only one row defined in the metadata since `full_response.content` has no `facet_fields`. Handling
this one-off case for now seems sufficient in my eyes since the way data_sets 2 filebrowser interacts with Solr doesn't produce this extra 400 error.

Fixes are encapsulated within #1479
